projects/ad400x/common/ad400x_bd.tcl

```diff
@@ -0,0 +1,119 @@
+
+  ad_ip_instance axi_spi_engine axi
+  ad_ip_parameter axi CONFIG.DATA_WIDTH $adc_resolution
+  ad_ip_parameter axi CONFIG.NUM_OFFLOAD 1
+
+  ad_ip_instance spi_engine_offload offload
+  ad_ip_parameter offload CONFIG.DATA_WIDTH $adc_resolution
+
+  ad_ip_instance spi_engine_interconnect interconnect
+  ad_ip_parameter interconnect CONFIG.DATA_WIDTH $adc_resolution
+
+  ad_ip_instance util_pulse_gen trigger_gen
+
+  ## to setup the sample rate of the system change the PULSE_PERIOD value
+  ## the acutal sample rate will be PULSE_PERIOD * (1/sys_cpu_clk)
+  ## fsys_cpu_clk must be 100 MHz
+  ## Default tCYC = 560 ns
+  ad_ip_parameter trigger_gen CONFIG.PULSE_PERIOD 56
+  ad_ip_parameter trigger_gen CONFIG.PULSE_WIDTH 1
+
+  if {$adc_resolution != 16} {
+    ad_ip_instance util_axis_upscale axis_upscaler
+    ad_ip_parameter axis_upscaler CONFIG.NUM_OF_CHANNELS 1
+    ad_ip_parameter axis_upscaler CONFIG.DATA_WIDTH 20
+    ad_ip_parameter axis_upscaler CONFIG.UDATA_WIDTH 32
+    ad_connect axis_upscaler/dfmt_enable GND
+    ad_connect axis_upscaler/dfmt_type GND
+    ad_connect axis_upscaler/dfmt_se GND
+  }
+
+  ad_connect axi/spi_engine_offload_ctrl0 offload/spi_engine_offload_ctrl
+  ad_connect offload/spi_engine_ctrl interconnect/s0_ctrl
+  ad_connect axi/spi_engine_ctrl interconnect/s1_ctrl
+  ad_connect interconnect/m_ctrl execution/ctrl
+
+  if {$adc_resolution != 16} {
+    ad_connect offload/offload_sdi axis_upscaler/s_axis
+    ad_connect axis_upscaler/m_axis M_AXIS_SAMPLE
+
+    ad_connect clk axis_upscaler/clk
+    ad_connect axi/spi_resetn axis_upscaler/resetn
+  } else {
+    ad_connect offload/offload_sdi M_AXIS_SAMPLE
+  }
+
+  ad_connect execution/spi m_spi
+
+  ad_connect clk offload/spi_clk
+  ad_connect clk offload/ctrl_clk
+  ad_connect clk execution/clk
+  ad_connect clk axi/s_axi_aclk
+  ad_connect clk axi/spi_clk
+  ad_connect clk interconnect/clk
+  ad_connect clk trigger_gen/clk
+
+  ad_connect axi/spi_resetn offload/spi_resetn
+  ad_connect axi/spi_resetn execution/resetn
+  ad_connect axi/spi_resetn interconnect/resetn
+  ad_connect axi/spi_resetn trigger_gen/rstn
+  ad_connect trigger_gen/pulse_period_en GND
+  ad_connect trigger_gen/pulse_period GND
+
+  ad_connect trigger_gen/pulse offload/trigger
+
+  ad_connect resetn axi/s_axi_aresetn
+
+current_bd_instance /
+
+# dma to receive data stream
+
+ad_ip_instance axi_dmac axi_ad400x_dma
+ad_ip_parameter axi_ad400x_dma CONFIG.DMA_TYPE_SRC 1
+ad_ip_parameter axi_ad400x_dma CONFIG.DMA_TYPE_DEST 0
+ad_ip_parameter axi_ad400x_dma CONFIG.CYCLIC 0
+ad_ip_parameter axi_ad400x_dma CONFIG.SYNC_TRANSFER_START 0
+ad_ip_parameter axi_ad400x_dma CONFIG.AXI_SLICE_SRC 0
+ad_ip_parameter axi_ad400x_dma CONFIG.AXI_SLICE_DEST 1
+ad_ip_parameter axi_ad400x_dma CONFIG.DMA_2D_TRANSFER 0
+
+if {$adc_resolution != 16} {
+  ad_ip_parameter axi_ad400x_dma CONFIG.DMA_DATA_WIDTH_SRC 32
+} else {
+  ad_ip_parameter axi_ad400x_dma CONFIG.DMA_DATA_WIDTH_SRC [expr $adc_resolution * 4]
```

Comment:
Why 4 times? Shouldn't it be fixed 16?

Reply:
Yes, it can and should be fixed. Thanks!